src/Components/Components/src/Services/IComponentContext.cs

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.AspNetCore.Components.Services
```

Comment:
What would you think about putting this in `Microsoft.AspNetCore.Components`. I really think we'll end up removing `.Services` as a namespace when this gets reviewed in favor of some mix of: 
- moving these types into `.Components`
- moving these types into more specific namespaces like routing/layouts
- moving them into a more generic namespace like `.Infrastructure` for things that most people won't use

Reply:
Totally happy to merge the `Components.Services` namespace into `.Components`. That would also fix one of the other things that's been bothering me lately - that the templates imports don't include `.Services`.

Filed https://github.com/aspnet/AspNetCore/issues/8892